math/mathcore/inc/Math/PdfFuncMathCore.h

```diff
@@ -524,15 +524,15 @@ namespace Math {
   inline double poisson_pdf(unsigned int n, double mu) {
     // Inlined to enable clad-auto-derivation for this function.
 
-    if (n > 0)
+    if (n > 0 && mu >= 0)
```

Comment:
`mu == 0` will lead to `std::log(0) = -inf` in the body. Was this intended?

Reply:
Maybe it was. `std::exp(std::log(0))` evaluates to 0.

Reply:
Yes, it was intended, we could add an extra if for mu == 0 but it is not needed